### DISCUSSION TITLE
Is it possible to switch between multiple layouts?

### DISCUSSION THREAD

**Discussion by btimofeev** [8VIM/8VIM#207] [Q&A]
Hello. Can you please tell me, is it possible to switch two languages? 

I would like to make a Russian layout, but we need to be able to switch to English too. The Russian language uses 33 Cyrillic letters (https://en.wikipedia.org/wiki/Russian_alphabet), in English there are 28 letters, they cannot be accommodated in one layout. 

Typically all keyboards have a button or gesture to switch between multiple selected layouts. Is this possible in 8Vim now, or will I have to implement this functionality myself?

**Comment by adaschma** [comment]
In the next release there will be a way to switch between the different layouts. Switching won't be terribly quick, however: requiring 6 clicks or so. Do you have a layout in mind? The key board only has 32 "front seat" spots. Some of the letters could be entered like diacritics are entered for other languages.

**Reply by btimofeev** [reply]
6 clicks is a lot.  Will the button be in the settings?  It should be displayed on the keyboard view.  There is a lot of free space on the screen.

**Comment by flide** [comment]
@btimofeev If you can find a way to make it work, please put up your suggestion and I will see what can be done.

If you can implement it... even better.

**Comment by balsoft** [comment]
Sadly this issue (no quick way to switch layouts) makes the keyboard practically useless for multilingual users :(

For now, I have set GBoard as the "emoji keyboard" and switch to it whenever I need to type in languages other than English.

It would be great to have a setting to add a simple button for layout switching to the left toolbar, perhaps replacing the emoji button?
